test/puppetlabs/trapperkeeper/services_test.clj

```diff
@@ -62,6 +64,67 @@
           app (bootstrap-services-with-empty-config [service1])]
       (is (not (nil? app))))))
 
+(deftest test-restart-file-correct-count
+  (testing "create a restart file and check that it correctly increments when a service starts"
+    (let [service1 (service Service1
+                            []
+                            (service1-fn [this] "hi"))
+          temp-file (kitchensink/temp-file-name "counter")]
```

Comment:
yes, `temp-file-name` does not create a file, only generates a name in tmp/ . Also temp-file did not delete the files off my system after the JVM stopped. very mysterious